{"abstract":[{"text":"A StreamError was hit during outbound frame processing.","type":"text"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/niohttp2\/niohttp2errors\/streamerror"]}],"schemaVersion":{"minor":3,"patch":0,"major":0},"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError"},"metadata":{"modules":[{"name":"NIOHTTP2"}],"symbolKind":"struct","title":"NIOHTTP2Errors.StreamError","role":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"StreamError","kind":"identifier"}],"navigatorTitle":[{"text":"StreamError","kind":"identifier"}],"externalID":"s:8NIOHTTP20A6ErrorsO11StreamErrorV","roleHeading":"Structure"},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors"]]},"topicSections":[{"title":"Instance Properties","identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/baseError","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/description","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/streamID"]}],"relationshipsSections":[{"kind":"relationships","title":"Conforms To","identifiers":["doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","doc:\/\/co.pointfree.VaporRouting\/s8SendableP"],"type":"conformsTo"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"StreamError"}],"platforms":["Linux"],"languages":["swift"]}]},{"kind":"content","content":[{"type":"heading","level":2,"text":"Overview","anchor":"overview"},{"type":"paragraph","inlineContent":[{"text":"Stream errors are wrappers around another error of some other kind that occurred on a specific stream.","type":"text"},{"text":" ","type":"text"},{"text":"As they are a wrapper error, they carry a “real” error in the ","type":"text"},{"type":"codeVoice","code":"baseError"},{"text":". Additionally, they cannot","type":"text"},{"text":" ","type":"text"},{"text":"meaningfully be ","type":"text"},{"type":"codeVoice","code":"Equatable"},{"text":", so they aren’t. There’s also no additional location information: that’s","type":"text"},{"text":" ","type":"text"},{"text":"provided by the base error.","type":"text"}]}]}],"sections":[],"references":{"doc://co.pointfree.VaporRouting/s5ErrorP":{"type":"unresolvable","title":"Swift.Error","identifier":"doc:\/\/co.pointfree.VaporRouting\/s5ErrorP"},"doc://co.pointfree.VaporRouting/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/co.pointfree.VaporRouting\/s8SendableP"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"StreamError"}],"role":"symbol","title":"NIOHTTP2Errors.StreamError","abstract":[{"type":"text","text":"A StreamError was hit during outbound frame processing."}],"navigatorTitle":[{"kind":"identifier","text":"StreamError"}],"type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror","kind":"symbol"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2":{"url":"\/documentation\/niohttp2","title":"NIOHTTP2","role":"collection","type":"topic","kind":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2","abstract":[]},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/streamID":{"fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"streamID"},{"kind":"text","text":": "},{"preciseIdentifier":"s:8NIOHTTP213HTTP2StreamIDV","kind":"typeIdentifier","text":"HTTP2StreamID"}],"role":"symbol","title":"streamID","abstract":[],"type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/streamID","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/streamid","kind":"symbol"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors":{"url":"\/documentation\/niohttp2\/niohttp2errors","title":"NIOHTTP2Errors","role":"symbol","type":"topic","fragments":[{"text":"enum","kind":"keyword"},{"text":" ","kind":"text"},{"text":"NIOHTTP2Errors","kind":"identifier"}],"kind":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors","abstract":[{"type":"text","text":"Errors that NIO raises when handling HTTP\/2 connections."}],"navigatorTitle":[{"text":"NIOHTTP2Errors","kind":"identifier"}]},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/baseError":{"type":"topic","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/baseerror","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/baseError","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"baseError","kind":"identifier"},{"text":": ","kind":"text"},{"text":"Error","kind":"typeIdentifier","preciseIdentifier":"s:s5ErrorP"}],"role":"symbol","title":"baseError","abstract":[],"kind":"symbol"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/description":{"role":"symbol","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/description","kind":"symbol","fragments":[{"kind":"keyword","text":"var"},{"kind":"text","text":" "},{"kind":"identifier","text":"description"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:SS","text":"String"}],"type":"topic","abstract":[],"title":"description","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/description"}}}